exec/cnex/number.c

```diff
@@ -510,8 +511,27 @@ BOOL number_is_odd(Number x)
 
 //#define __NUMBER_TESTS
 #ifdef __NUMBER_TESTS
+#include "time.h"
 void main()
 {
+     srand((unsigned int)time(NULL));
+
+    for (int i = 0; i < 100000; i++) {
+        BID_UINT128 n = { 0 };
+        n.w[0] = rand();
```

Comment:
This really isn't going to do what you want here. I'm pretty sure that `rand()` on your platform returns a 32-bit number, but `w[0]` is a 64 bit value. You'll want to make sure that you fill *all* the bits of `n` with random data, not just half of them.

Reply:
I checked it, it just returns an int, so under amd64 it works, but I didn't think of making sure it works under x86 also.  Originally, I thought it was 32 bit also, and I tried to do a shift and an or, but that was when I realized it wouldn't let me shift by 32 bits.  I presume I can just do some simple sizeof() checks on what rand() returns, and if it is 4, then do a left sift by 32 bits, and then call rand() again and logically or it with the previous result.  I was also monitoring the numbers I was formatting, and they were some pretty wacky values.  It might be beneficial to have some smaller numbers to format too.

Reply:
I don't think this has anything to do with amd64 vs x86. The import point is, what is `RAND_MAX` on your platform? I would be surprised if it returns a 64 bit random number.

Reply:
Fixed with a platform independent method.